File: php7/src/ChainOfResponsibility/Tests/Agent/LineManagerTest.php
<?php

declare(strict_types = 1);

namespace PetrePatrasc\ChainOfResponsibility\Tests\Agent;


use PetrePatrasc\ChainOfResponsibility\Agent\AbstractLoanAgent;
use PetrePatrasc\ChainOfResponsibility\Agent\LineManager;

class LineManagerTest extends AbstractLoanAgentTest
{
    /**
     * @var LineManager
     */
    protected $lineManager;

    /**
     * @inheritDoc
     */
    protected function setUp()
    {
        parent::setUp();

        $this->lineManager = new LineManager();
    }

    /**
     * @inheritDoc
     */
    public function loanAmountDataProvider(): array
    {
        return [
            [300.17, true],
            [9474.50, true],
            [15000, true],
            [19999.99, true],
            [20000, false],
        ];
    }

    /**
     * @inheritDoc
     */
    protected function getTestedAgent(): AbstractLoanAgent
    {
        return $this->lineManager;
    }
}

File: java/src/ro/petrepatrasc/chainofresponsability/tests/agent/LineManagerTest.java
package ro.petrepatrasc.chainofresponsability.tests.agent;

import com.tngtech.java.junit.dataprovider.DataProvider;
import ro.petrepatrasc.chainofresponsability.agent.AbstractLoanAgent;
import ro.petrepatrasc.chainofresponsability.agent.BranchManager;
import ro.petrepatrasc.chainofresponsability.agent.LineManager;

public class LineManagerTest extends AbstractLoanAgentTest {

    @Override
    public AbstractLoanAgent getTestedAgent() {
        return new LineManager();
    }

    @DataProvider
    public static Object[][] loanAmountDataProvider() {
        return new Object[][]{
                {9371.137, true},
                {18360.28, true},
                {15120.75, true},
                {13985.48, true},
                {19999.99, true},
                {20000, false},
                {25000.127, false},
                {27158.89, false},
        };
    }
}
